src/_pytest/config/__init__.py

```diff
@@ -1339,6 +1339,7 @@ def create_terminal_writer(config: Config, *args, **kwargs) -> TerminalWriter:
     tw = TerminalWriter(*args, **kwargs)
     if config.option.color == "yes":
         tw.hasmarkup = True
+    tw.code_highlight = config.option.code_highlight == "yes" and tw.hasmarkup
     if config.option.color == "no":
         tw.hasmarkup = False
```

Comment:
I suggest a few changes:

- Move the `code_highlight` below the last two lines otherwise it cuts in the middle of the `color` yes/no logic
- Make `code_highlight` boolean itself independent of `hasmarkup` -- the highlight code already checks `if not self.hasmarkup or not self.code_highlight` so it is better to let it standalone just in case.
- If no an explicit yes/no isn't given, don't override the default (same logic as `color` uses)

```suggestion

    if config.option.color == "yes":
        tw.hasmarkup = True
    elif config.option.color == "no":
        tw.hasmarkup = False

    if config.option.code_highlight == "yes":
        tw.code_highlight =  True
    elif config.option.code_highlight == "no":
        tw.code_highlight =  False
```